x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -67,6 +77,39 @@ public FieldAttribute(
         this.field = field;
     }
 
+    @SuppressWarnings("unchecked")
+    public <S extends StreamInput & PlanStreamInput> FieldAttribute(StreamInput in) throws IOException {
+        this(
+            Source.readFrom((S) in),
```

Comment:
This funny casting dance is required because we're in esql-core here and the real `PlanStreamInput` is in esql-proper. And because `NamedWriteableRegistry.Entry` needs ` StreamInput`, not a `PlanStreamInput`. And we need `PlanStreamInput` to handle Source and NameId.

Reply:
This PR comment should be a code comment.  Because in six months when I'm scratching my head to figure out why we did it this way, I'll never find this comment.

Reply:
Sure.

Reply:
a82a5d31d0a